BetaCameras/Kinect4Azure/Kinect4Azure.cs

```diff
@@ -0,0 +1,531 @@
+
+            if (disposing)
+            {
+                // dispose managed resources
+            }
+
+            _disposed = true;
+        }
+
+        protected override void LoadAllAvailableChannels()
+        {
+            ChannelRegistry cr = ChannelRegistry.Instance;
+            Channels.Clear();
+
+            Channels.Add(cr.RegisterChannel(ChannelNames.ZImage));
+            Channels.Add(cr.RegisterChannel(ChannelNames.Distance));
+            Channels.Add(cr.RegisterChannel(ChannelNames.Color));
+            Channels.Add(cr.RegisterChannel(ChannelNames.Intensity));
+        }
+
+        protected override void ConnectImpl()
+        {
+            bool haveSerial = !string.IsNullOrWhiteSpace(SerialNumber);
+
+            if (!haveSerial)
+            {
+                _device = Device.Open(0);
+                this.SerialNumber = _device.SerialNum;
+            }
+            else
+            {
+                for(int i = 0; i < Device.GetInstalledCount(); i++)
+                {
+                    Device tmpDev = Device.Open(i);
+                    if (SerialNumber == tmpDev.SerialNum)
+                    {
+                        _device = tmpDev;
+                        break;
+                    }
+                    tmpDev.Dispose();
+                }
+            }
+
+            if (ActiveChannels.Count == 0)
+            {
+                AddToActiveChannels(ChannelNames.Color);
+                AddToActiveChannels(ChannelNames.ZImage);
+                AddToActiveChannels(ChannelNames.Distance);
+                AddToActiveChannels(ChannelNames.Intensity);
+            }
+
+            RestartCamera();
+        }
+
+        private void RestartCamera()
+        {
+            _reset.Reset();
+            _device.StopCameras();
+            if (DepthMode == K4ADepthMode.WFOV_Unbinned)
+            {
+                _fps = FPS.fps15;
+            }
+            else
+            {
+                _fps = FPS.fps30;
+            }
+
+            bool synchronizedImagesOnly = _colorResolution != Microsoft.AzureKinect.ColorResolution.Off && _depthMode != Microsoft.AzureKinect.DepthMode.Off; //Off-mode does not support synchronization
+
+            _device.StartCameras(new DeviceConfiguration
+            {
+                ColorFormat = Microsoft.AzureKinect.ImageFormat.ColorBGRA32,
+                ColorResolution = _colorResolution,
+                DepthMode = _depthMode,
+                SynchronizedImagesOnly = synchronizedImagesOnly,
+                CameraFPS = _fps,
+            });
+            _reset.Set();
+        }
+
+        protected override void DisconnectImpl()
+        {
+            _intrinsicsCache.Clear();
+            _extrinsicsCache.Clear();
+
+            if (null != _device)
+            {
+                _device.StopCameras();
+                _device.Dispose();
+            }
+        }
+
+        protected override void UpdateImpl()
+        {
+            _reset.WaitOne();
```

Comment:
Is the field `_reset` only used for synchronization? In that case I see problems if `RestartCamera` is called directly after the `WaitOne` line.